File: firstspringdemo/src/main/java/firstspring/firstspringdemo/DTOs/ProductUpdateDto.java
package firstspring.firstspringdemo.DTOs;

public class ProductUpdateDto {
    private String name;
    private String description;
    private double price;
    private int quantity;
    private String locationId;

    public ProductUpdateDto() {
    }

    public ProductUpdateDto(String name, String description, double price, int quantity, String locationId) {
        this.name = name;
        this.description = description;
        this.price = price;
        this.quantity = quantity;
        this.locationId = locationId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public String getLocationId() {
        return locationId;
    }

    public void setLocationId(String locationId) {
        this.locationId = locationId;
    }
}

File: firstspringdemo/flutter/lib/models/product_update_dto.dart
class ProductUpdateDto {
  final String? name;
  final String? description;
  final double? price;
  final int? quantity;
  final String? locationId;

  ProductUpdateDto({
    this.name,
    this.description,
    this.price,
    this.quantity,
    this.locationId,
  });

  factory ProductUpdateDto.fromJson(Map<String, dynamic> json) {
    return ProductUpdateDto(
      name: json['name'],
      description: json['description'],
      price: json['price']?.toDouble(),
      quantity: json['quantity'],
      locationId: json['locationId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'price': price,
      'quantity': quantity,
      'locationId': locationId,
    };
  }
}
